pkg/errors/armerrors.go

```diff
@@ -44,13 +44,20 @@ func ZonalAllocationFailureOccurred(err error) bool {
 	return azErr != nil && azErr.ErrorCode == ZoneAllocationFailed
 }
 
+// SKUFamilyQuotaHasBeenReached tells us if we have exceeded our Quota.
+func SKUFamilyQuotaHasBeenReached(err error) bool {
+	azErr := IsResponseError(err)
+	return azErr != nil && azErr.ErrorCode == OperationNotAllowed && strings.Contains(azErr.Error(), SKUFamilyQuotaExceededTerm)
```

Comment:
I very much don't like string matching, but assuming there isn't a field. ErrorSubcode or anything else we can look at?

Reply:
Assuming there is nothing else we can inspect for this info it seems ok.

Reply:
Nope, we don't have any additional info and the error messages content isn't easily marshaled :( 


This is unfortunately the best way we can do this with the current sdk errors returned, as they are optimized to be human readable and not machine readable.

Reply:
@Bryce-Soghigian  - so having been living amongst RP errors for a while now, there are a few different error codes that will be set for core quota issues.

union withsource=TS FrontEndQoSEvents, AsyncQoSEvents
| where PreciseTimeStamp > ago(10d)
| where errorDetails has_all ('quota', 'exceeding approved', 'Family Cores quota')
| summarize count() by TS, resultCode, resultSubCode
| order by count_ desc

TS | resultCode | resultSubCode | count_
-- | -- | -- | --
AsyncQoSEvents | QuotaExceeded | QuotaExceeded | 57
FrontEndQoSEvents | PreconditionFailed |   | 50
FrontEndQoSEvents | QuotaExceeded |   | 13


<p class="cs95E872D0"><span class="cs1B16EEB5">&nbsp;</span></p><!--EndFragment--></BODY></HTML>

Reply:
Thanks for sharing,yep seen these. We need to move these errors into karpenter slowly but surely